src/app/chat/view.nim

```diff
@@ -461,6 +461,13 @@ QtObject:
       self.messageList[chatId].delete
       self.messageList.del(chatId)
 
+  proc toggleReaction*(self: ChatsView, messageId: string, emojiId: int) {.slot.} =
+    if self.activeChannel.id == status_utils.getTimelineChatId():
+      let message = self.messageList[status_utils.getTimelineChatId()].getMessageById(messageId)
+      self.reactions.toggle(messageId, message.chatId, emojiId)
```

Comment:
Here we determine the "real" chat id of the message from which the reaction is coming from. This is needed because otherwise the reaction is sent to a wrong chat.